lib/ansible/module_utils/aws/rds.py

```diff
@@ -0,0 +1,168 @@
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2017 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+# Author: Michael De La Rue 2017 largely rewritten but based on work
+# by Will Thames taken in turn from the original rds module.
+
+try:
+    import botocore
+    from botocore import xform_name
+except:
+    pass  # it is assumed that calling modules will detect and provide an appropriate nice error.
+
+from ansible.module_utils.ec2 import camel_dict_to_snake_dict, boto3_tag_list_to_ansible_dict
+
+DEFAULT_PORTS = {
+    'aurora': 3306,
+    'mariadb': 3306,
+    'mysql': 3306,
+    'oracle': 1521,
+    'sqlserver': 1433,
+    'postgres': 5432,
+}
+
+DB_ENGINES = [
+    'aurora',
+    'mariadb',
+    'mysql',
+    'oracle-ee',
+    'oracle-se',
+    'oracle-se1',
+    'oracle-se2',
+    'postgres',
+    'sqlserver-ee',
+    'sqlserver-ex',
+    'sqlserver-se',
+    'sqlserver-web',
+]
+
+LICENSE_MODELS = [
+    'bring-your-own-license',
+    'general-public-license',
+    'license-included',
+    'postgresql-license'
+]
+
+
+def get_db_instance(conn, instancename):
+    """return AWS format DB instance
+
+    This function connects to AWS and retrieves the information about
+    the instance in directly in the standard AWS format.
+    """
+    # FIXME: currently this makes two API calls, one for the instance and
+    # one for the tags.  This mostly make sense since the object we get
+    # back is close to what we want to call a create with, but for
+    # efficiency we should maybe put an option to not make the tags call in
+    # the case where we throw that data away.
+    try:
+        response = conn.describe_db_instances(DBInstanceIdentifier=instancename)
+    except botocore.exceptions.ClientError as e:
+        if e.response['Error']['Code'] == 'DBInstanceNotFound':
+            return None
+        raise
+
+    instance = response['DBInstances'][0]
+
+    tags = conn.list_tags_for_resource(ResourceName=instance['DBInstanceArn']).get('TagList', [])
+    instance['Tags'] = boto3_tag_list_to_ansible_dict(tags)
+
+    return instance
+
+
+def instance_to_facts(instance):
+    if not instance:
+        return instance
+    if 'DBInstanceIdentifier' not in instance:
+        raise AssertionError("instance argument was not a valid instance")
+    return camel_dict_to_snake_dict(instance, ignore_list=['Tags'])
+
+
+def get_snapshot(conn, snapshotid):
+    try:
+        response = conn.describe_db_snapshots(DBSnapshotIdentifier=snapshotid)
+    except botocore.exceptions.ClientError as e:
```

Comment:
That is nice! Is that documented anywhere? Presumably I should still have the `except (ClientError, BotoCoreError) as e` for actual errors?

Reply:
It's in the boto3 docs, but we don't have it in guidelines (guess I should do that, oops). In this case since you're re-raising, you don't need to also catch ClientError since this would just re-raise that, so only the NotFound check is fine here. In general, yes you still need `except (ClientError, BotoCoreError) as e` since this only catches that single subtype.